include/fmt/base.h

```diff
@@ -427,6 +427,33 @@ enum class uint128_opt {};
 template <typename T> auto convert_for_visit(T) -> monostate { return {}; }
 #endif
 
+#ifndef FMT_USE_BITINT
+#  if FMT_CLANG_VERSION >= 1400
+#    define FMT_USE_BITINT 1
+#  else
+#    define FMT_USE_BITINT 0
+#  endif
+#endif
+
+template <class T, int N = 0> struct bitint_traits : std::false_type {};
+#if FMT_USE_BITINT
+#  pragma clang diagnostic push
+#  pragma clang diagnostic ignored "-Wbit-int-extension"
+
+// fmt only supports up to 128 bits https://github.com/fmtlib/fmt/pull/4072
+template <int N> struct bitint_traits<_BitInt(N)> : std::true_type {
+  static constexpr bool is_signed = true;
+  static constexpr size_t number_of_bits = N;
```

Comment:
Can we use numeric_limits instead?

Reply:
> Can we use numeric_limits instead?

Not if we want this to work when using clang with libstdc++. The libstdc++ implementation just has a load of specialisations for deciding what is signed, and `_BitInt` isn't in it, so it doesn't behave correctly - see https://godbolt.org/z/86hsr5dqK

This works fine with libc++ (which uses an internal function)

Thus I'd say best to leave this as is